pandas/tests/extension/base/interface.py

```diff
@@ -29,6 +29,31 @@ def test_can_hold_na_valid(self, data):
         # GH-20761
         assert data._can_hold_na is True
 
+    def test_contains(self, data, data_missing):
+        # GH-37867
+        # Tests for membership checks. Membership checks for nan-likes is tricky and
+        # the settled on rule is: `nan_like in arr` is True if nan_like is
+        # arr.dtype.na_value and arr.isna().any() is True. Else the check returns False.
+
+        na_value = data.dtype.na_value
+        # ensure data without missing values
+        data = data[~data.isna()]
+
+        # first elements are non-missing
+        assert data[0] in data
+        assert data_missing[0] in data_missing
+
+        # check the presence of na_value
+        assert na_value in data_missing
+        assert na_value not in data
+
+        # the data can never contain other nan-likes than na_value
```

Comment:
might consider actually using the nulls_fixture as this is more comprehensive (sure it will run the other checks multiple times but no big deal).

Reply:
I am personally -1 on using a fixture for this (we could move the list of nulls that is currently used for defining the fixture to constant in `_testing.py` and use that constant in both places, though)

Reply:
in this case?

we already have a nulls fixture and use it in a great many places

we need to be comprehensive and general in testing - specific cases are ok sometimes

Reply:
ok here's the fixture contents
```
@pytest.fixture(params=[None, np.nan, pd.NaT, float("nan"), pd.NA], ids=str)
```

so if you add `float('nan')` then this should cover

Reply:
I like the generality of using the nulls_fixture too. I've added it in the newest commit.